virtual_ecosystem/core/registry.py

```diff
@@ -149,6 +110,63 @@ def register_module(module_name: str) -> None:
     )
 
 
+def get_model(module_name: str, module_name_short: str):
+    """Get the model class for a model.
+
+    Discovery is by looking for a single BaseModel subclass within a submodule with a
+    name matching the the model short name plus '_model'.
+
+    * ``models.plants`` -> ``models.plants.plants_model.py``,
+
+    Args:
+        module_name: The full module name (e.g. ``virtual_ecosystem.models.plants``)
+        module_name_short: The short module name (e.g ``plants``)
+    """
+
+    from virtual_ecosystem.core.base_model import BaseModel
+
+    # Try and import the module from the name to get a reference to the module
+    try:
+        module = import_module(module_name + f".{module_name_short}_model")
+    except ModuleNotFoundError as excep:
+        LOGGER.critical(f"Unknown module - registration failed: {module_name}")
+        raise excep
+
+    # Locate _one_ BaseModel subclass in the module that is not the imported
+    # BaseModelABC.
+    models_found = [
+        (obj_name, obj)
+        for obj_name, obj in getmembers(module)
+        if isclass(obj) and issubclass(obj, BaseModel) and obj is not BaseModel
+    ]
+
+    # Trap missing and multiple models
+    if len(models_found) == 0:
+        msg = f"Model object not found in {module_name}"
+        LOGGER.critical(msg)
+        raise RuntimeError(msg)
+
+    if len(models_found) > 1:
+        msg = f"More than one model defined in in {module_name}"
+        LOGGER.critical(msg)
+        raise RuntimeError(msg)
```

Comment:
Wasn't the case that all models follow the name convention that if the model short name is `my_model_name` then the model calss is `MyModelNameModel`? If that's the case, and as done with the pydantic configuration classes, this whole thing can be reduced a lot.

Reply:
Agreed - I've updated it to look for `model_name.model_name_model.ModelNameModel`